### PR TITLE
Published container not working with social-app-template

### PR DESCRIPTION
**Pull request by mattheworris**
The `content-watcher-service` fails to execute when used with `social-app-template` because it can't find `/usr/bin/tini`.

## Solution:

`tini` on Alpine lives at `/sbin/tini`.

This results in the following error:

```bash
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [NestFactory] Starting Nest application...
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] EventEmitterModule dependencies initialized +22ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ConfigHostModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +1ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardFeatureModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullBoardRootModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ScheduleModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] EventEmitterModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +1ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BlockchainModule dependencies initialized +2ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +3ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] BullModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] RedisModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] IPFSProcessorModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ScannerModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] CrawlerModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] PubSubModule dependencies initialized +0ms
2024-07-02 16:22:04 [Nest] 7  - 07/02/2024, 10:22:04 PM     LOG [InstanceLoader] ApiModule dependencies initialized +0ms
2024-07-02 16:22:04 /app/node_modules/@nestjs/bullmq/dist/hosts/worker-host.class.js:7
2024-07-02 16:22:04             throw new Error('"Worker" has not yet been initialized. Make sure to interact with worker instances after the "onModuleInit" lifecycle hook is triggered, for example, in the "onApplicationBootstrap" hook.');
2024-07-02 16:22:04                   ^
2024-07-02 16:22:04 
2024-07-02 16:22:04 Error: "Worker" has not yet been initialized. Make sure to interact with worker instances after the "onModuleInit" lifecycle hook is triggered, for example, in the "onApplicationBootstrap" hook.
2024-07-02 16:22:04     at get worker [as worker] (/app/node_modules/@nestjs/bullmq/dist/hosts/worker-host.class.js:7:19)
2024-07-02 16:22:04     at CrawlerService.onModuleDestroy (/app/dist/apps/api/main.js:2602:20)
2024-07-02 16:22:04     at MapIterator.iteratee (/app/node_modules/@nestjs/core/hooks/on-module-destroy.hook.js:22:43)
2024-07-02 16:22:04     at MapIterator.next (/app/node_modules/iterare/lib/map.js:13:39)
2024-07-02 16:22:04     at IteratorWithOperators.next (/app/node_modules/iterare/lib/iterate.js:21:28)
2024-07-02 16:22:04     at Function.from (<anonymous>)
2024-07-02 16:22:04     at IteratorWithOperators.toArray (/app/node_modules/iterare/lib/iterate.js:180:22)
2024-07-02 16:22:04     at callOperator (/app/node_modules/@nestjs/core/hooks/on-module-destroy.hook.js:23:10)
2024-07-02 16:22:04     at callModuleDestroyHook (/app/node_modules/@nestjs/core/hooks/on-module-destroy.hook.js:43:23)
2024-07-02 16:22:04     at NestApplication.callDestroyHook (/app/node_modules/@nestjs/core/nest-application-context.js:233:53)
2024-07-02 16:22:04 
2024-07-02 16:22:04 Node.js v20.12.2
```

## Changes:
- `Dockerfile` rolled back to Node:Ubuntu/Debian base image
- Add environment to `docker-compose.yaml`

### CODE REVIEW

**Review by saraswatpuneet — APPROVED** [review]

File: docker-compose.yaml [hunk 1]
```diff
@@ -79,15 +100,15 @@ services:
     image: content-watcher-service
     build:
       context: .
-      dockerfile: dev.Dockerfile
+      dockerfile: Dockerfile
       tags:
         - content-watcher-service:latest
-    env_file:
-      - .env.docker.dev
```
Comment (saraswatpuneet) [inline docker-compose.yaml:86]: nit: is it not preferred to use a file for env?

File: docker-compose.yaml [hunk 2]
```diff
@@ -79,15 +100,15 @@ services:
     image: content-watcher-service
     build:
       context: .
-      dockerfile: dev.Dockerfile
+      dockerfile: Dockerfile
```
Comment (saraswatpuneet) [inline docker-compose.yaml:103]: nit: just that dev.Dockerfile would have rebuilt project on changes to code files, but i guess a tradeoff between making it work for mac/linux

File: docker-compose.yaml (hunk 1 above)
Comment (mattheworris) [reply docker-compose.yaml:86]: In gateway, we moved the env files into the docker-compose and seems like it's easier to manage that way.

File: docker-compose.yaml (hunk 2 above)
Comment (mattheworris) [reply docker-compose.yaml:103]: Yeah, I don't know if I want to commit this change, but it did sort of hide the fact that the container wasn't working, so we might need 2 versions, one for development and one for testing the production image. 🤔

Comment (JoeCap08055) [reply docker-compose.yaml:103]: dev.Dockerfile works fine for me. The point is, it's only for local development. Assuming your container & local environment are the same architecture, you have to manually manage your node_modules folder outside of the container.

File: Dockerfile [hunk 3]
```diff
@@ -1,72 +1,33 @@
...
+	apt-get install -y jq curl tini && \
+	apt-get clean && \
+	rm -rf /usr/share/doc /usr/share/man /usr/share/zsh
 
-# Expose the port that the application listens on.
 EXPOSE 3000
 
-# Run the application.
-ENTRYPOINT ["/usr/bin/tini", "--", "node", "dist/apps/api/main.js"]
+ENTRYPOINT ["/usr/bin/tini", "--", "npm", "run", "start:api:prod"]
```
Comment (JoeCap08055) [inline Dockerfile:33]: Don't we want to prefer running node directly on built JS output, instead of using `npm run ...`?

**Pull request merged**